// elevate-dynamic-form/esm2022/lib/components/star-rating/star-rating.component.mjs
import { Component, Input } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/material/icon";
export class StarRatingComponent {
    constructor() {
        this.rating = 0;
        this.touched = false;
        this.onChange = (change) => { };
        this.onTouched = () => { };
    }
    ngOnInit() {
        this.ratingList = [];
        for (let i = 0; i < this.starsCount; i++) {
            this.ratingList.push(i);
        }
    }
    writeValue(value) {
        this.rating = value;
    }
    registerOnChange(onChange) {
        this.onChange = onChange;
    }
    registerOnTouched(onTouched) {
        this.onTouched = onTouched;
    }
    markAsTouched() {
        if (!this.touched) {
            this.onTouched();
            this.touched = true;
        }
    }
    onRatingChange(rate) {
        this.markAsTouched();
        this.rating = rate + 1;
        this.onChange(this.rating);
    }
    displayIcons(idx) {
        return this.rating && this.rating >= idx + 1 ? 'star' : 'star_border';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: StarRatingComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.8", type: StarRatingComponent, selector: "lib-star-rating", inputs: { starsCount: "starsCount", label: "label" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                multi: true,
                useExisting: StarRatingComponent,
            },
        ], ngImport: i0, template: "<div class=\"rating-list\">\n  <div *ngFor=\"let star of ratingList; let idx = index\">\n    <mat-icon class=\"rating-icon\" (click)=\"onRatingChange(idx)\">{{\n      displayIcons(idx)\n    }}</mat-icon>\n  </div>\n</div>\n", styles: [".rating-list{display:flex;flex-direction:row;margin-top:10px}.rating-icon{font-size:30px;margin-right:10px;color:gold;cursor:pointer}\n"], dependencies: [{ kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "component", type: i2.MatIcon, selector: "mat-icon", inputs: ["color", "inline", "svgIcon", "fontSet", "fontIcon"], exportAs: ["matIcon"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: StarRatingComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-star-rating', providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            multi: true,
                            useExisting: StarRatingComponent,
                        },
                    ], template: "<div class=\"rating-list\">\n  <div *ngFor=\"let star of ratingList; let idx = index\">\n    <mat-icon class=\"rating-icon\" (click)=\"onRatingChange(idx)\">{{\n      displayIcons(idx)\n    }}</mat-icon>\n  </div>\n</div>\n", styles: [".rating-list{display:flex;flex-direction:row;margin-top:10px}.rating-icon{font-size:30px;margin-right:10px;color:gold;cursor:pointer}\n"] }]
        }], ctorParameters: () => [], propDecorators: { starsCount: [{
                type: Input
            }], label: [{
                type: Input
            }] } });
