// elevate-dynamic-form/esm2022/lib/components/radio-input/radio-input.component.mjs
import { Component, Input } from '@angular/core';
import { FormControl, NG_VALUE_ACCESSOR, } from '@angular/forms';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@angular/common";
import * as i3 from "@angular/material/radio";
export class RadioInputComponent {
    constructor(fb) {
        this.fb = fb;
        this.touched = false;
        this.stateChanges = new Subject();
        this.onChange = (value) => { };
        this.onTouched = () => { };
        this.form = this.fb.group({});
    }
    ngOnInit() {
        this.form.addControl(this.radioControl.name, new FormControl(this.radioControl.value || null));
        this.form.get(this.radioControl.name)?.valueChanges.subscribe((value) => {
            this.onChange(value);
            this.stateChanges.next();
        });
    }
    writeValue(value) {
        this.selectedValue = '';
    }
    registerOnChange(onChange) {
        this.onChange = onChange;
    }
    registerOnTouched(onTouched) {
        this.onTouched = onTouched;
    }
    markAsTouched() {
        if (!this.touched) {
            this.onTouched();
            this.touched = true;
        }
    }
    ngOnDestroy() {
        this.stateChanges.complete();
    }
    onRadioChange(event) {
        this.markAsTouched();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: RadioInputComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.8", type: RadioInputComponent, selector: "lib-radio-input", inputs: { radioControl: "radioControl" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: RadioInputComponent,
                multi: true,
            },
        ], ngImport: i0, template: "<div [formGroup]=\"form\" style=\"margin-top: 10px\">\n  <mat-radio-group\n    [value]=\"radioControl?.value\"\n    [formControlName]=\"radioControl?.name\"\n    [required]=\"radioControl?.validators?.required\"\n    (change)=\"onRadioChange($event)\"\n    class=\"radio-options-list\"\n  >\n    <mat-radio-button\n      *ngFor=\"let option of radioControl?.options\"\n      [value]=\"option.value\"\n      style=\"margin-bottom: 10px\"\n      >{{ option?.label }}</mat-radio-button\n    >\n  </mat-radio-group>\n</div>\n", styles: [".radio-options-list{display:flex;flex-direction:column}\n"], dependencies: [{ kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.RequiredValidator, selector: ":not([type=checkbox])[required][formControlName],:not([type=checkbox])[required][formControl],:not([type=checkbox])[required][ngModel]", inputs: ["required"] }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i3.MatRadioGroup, selector: "mat-radio-group", inputs: ["color", "name", "labelPosition", "value", "selected", "disabled", "required"], outputs: ["change"], exportAs: ["matRadioGroup"] }, { kind: "component", type: i3.MatRadioButton, selector: "mat-radio-button", inputs: ["id", "name", "aria-label", "aria-labelledby", "aria-describedby", "disableRipple", "tabIndex", "checked", "value", "labelPosition", "disabled", "required", "color"], outputs: ["change"], exportAs: ["matRadioButton"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: RadioInputComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-radio-input', providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: RadioInputComponent,
                            multi: true,
                        },
                    ], template: "<div [formGroup]=\"form\" style=\"margin-top: 10px\">\n  <mat-radio-group\n    [value]=\"radioControl?.value\"\n    [formControlName]=\"radioControl?.name\"\n    [required]=\"radioControl?.validators?.required\"\n    (change)=\"onRadioChange($event)\"\n    class=\"radio-options-list\"\n  >\n    <mat-radio-button\n      *ngFor=\"let option of radioControl?.options\"\n      [value]=\"option.value\"\n      style=\"margin-bottom: 10px\"\n      >{{ option?.label }}</mat-radio-button\n    >\n  </mat-radio-group>\n</div>\n", styles: [".radio-options-list{display:flex;flex-direction:column}\n"] }]
        }], ctorParameters: () => [{ type: i1.FormBuilder }], propDecorators: { radioControl: [{
                type: Input
            }] } });
