// elevate-dynamic-form/esm2022/lib/dynamic-form.module.mjs
import { NO_ERRORS_SCHEMA, NgModule } from '@angular/core';
import { DynamicFormComponent } from './dynamic-form.component';
import { MainFormComponent } from './components/main-form/main-form.component';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatInputModule } from '@angular/material/input';
import { MatButtonModule } from '@angular/material/button';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatIconModule } from '@angular/material/icon';
import { MatSelectModule } from '@angular/material/select';
import { MatChipsModule } from '@angular/material/chips';
import { MatSlideToggleModule } from '@angular/material/slide-toggle';
import { MatDialogModule } from '@angular/material/dialog';
import { MatSliderModule } from '@angular/material/slider';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatRadioModule } from '@angular/material/radio';
import { ChipInputComponent } from './components/chip-input/chip-input.component';
import { DialogInputComponent } from './components/dialog-input/dialog-input.component';
import { RadioInputComponent } from './components/radio-input/radio-input.component';
import { CheckboxInputComponent } from './components/checkbox-input/checkbox-input.component';
import { StarRatingComponent } from './components/star-rating/star-rating.component';
import { MatDatepickerModule } from '@angular/material/datepicker';
import * as i0 from "@angular/core";
// import {
//   NgxMatDatetimePickerModule,
//   NgxMatTimepickerModule,
// } from '@angular-material-components/datetime-picker';
// import { NgxMatMomentModule } from '@angular-material-components/moment-adapter';
export class DynamicFormModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: DynamicFormModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.8", ngImport: i0, type: DynamicFormModule, declarations: [DynamicFormComponent,
            MainFormComponent,
            ChipInputComponent,
            DialogInputComponent,
            RadioInputComponent,
            CheckboxInputComponent,
            StarRatingComponent], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            // BrowserAnimationsModule,
            MatInputModule,
            MatButtonModule,
            MatFormFieldModule,
            MatIconModule,
            MatSelectModule,
            MatChipsModule,
            MatSlideToggleModule,
            MatDialogModule,
            MatSliderModule,
            MatCheckboxModule,
            MatRadioModule,
            MatDatepickerModule], exports: [MainFormComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: DynamicFormModule, imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            // BrowserAnimationsModule,
            MatInputModule,
            MatButtonModule,
            MatFormFieldModule,
            MatIconModule,
            MatSelectModule,
            MatChipsModule,
            MatSlideToggleModule,
            MatDialogModule,
            MatSliderModule,
            MatCheckboxModule,
            MatRadioModule,
            MatDatepickerModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: DynamicFormModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        DynamicFormComponent,
                        MainFormComponent,
                        ChipInputComponent,
                        DialogInputComponent,
                        RadioInputComponent,
                        CheckboxInputComponent,
                        StarRatingComponent,
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        // BrowserAnimationsModule,
                        MatInputModule,
                        MatButtonModule,
                        MatFormFieldModule,
                        MatIconModule,
                        MatSelectModule,
                        MatChipsModule,
                        MatSlideToggleModule,
                        MatDialogModule,
                        MatSliderModule,
                        MatCheckboxModule,
                        MatRadioModule,
                        MatDatepickerModule,
                        // NgxMatDatetimePickerModule,
                        // NgxMatTimepickerModule,
                        // NgxMatMomentModule,
                    ],
                    exports: [MainFormComponent],
                    schemas: [NO_ERRORS_SCHEMA]
                }]
        }] });
