// elevate-dynamic-form/esm2022/lib/components/checkbox-input/checkbox-input.component.mjs
import { Component, Input } from '@angular/core';
import { FormArray, FormControl, NG_VALUE_ACCESSOR, } from '@angular/forms';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
import * as i2 from "@angular/common";
import * as i3 from "@angular/material/checkbox";
export class CheckboxInputComponent {
    constructor(fb) {
        this.fb = fb;
        this.touched = false;
        this.form = this.fb.group({});
        this.stateChanges = new Subject();
        this.onChange = (value) => { };
        this.onTouched = () => { };
    }
    ngOnInit() {
        let value = this.checkboxControl.value;
        let optionsControl = this.checkboxControl.options.map((data) => {
            if (value &&
                value.find((val) => {
                    return val.label == data.label;
                })) {
                return new FormControl(data);
            }
            else {
                return new FormControl('');
            }
        });
        this.form.addControl(this.checkboxControl.name, new FormArray(optionsControl));
        this.form
            .get(this.checkboxControl.name)
            ?.valueChanges.subscribe((value) => {
            let selectedValues = value.filter((val) => {
                return val;
            });
            if (!selectedValues.length) {
                this.onChange('');
            }
            else {
                this.onChange(selectedValues);
            }
            this.stateChanges.next();
        });
    }
    writeValue(value) {
        this.selectedOptions = '';
    }
    registerOnChange(onChange) {
        this.onChange = onChange;
    }
    registerOnTouched(onTouched) {
        this.onTouched = onTouched;
    }
    markAsTouched() {
        if (!this.touched) {
            this.onTouched();
            this.touched = true;
        }
    }
    onCheckboxChange(event, option, index) {
        let currentControl = this.form.get(this.checkboxControl.name);
        if (event.checked) {
            currentControl.controls[index].patchValue(option);
        }
        else {
            currentControl.controls[index].patchValue('');
        }
        this.markAsTouched();
    }
    ngOnDestroy() {
        this.stateChanges.complete();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: CheckboxInputComponent, deps: [{ token: i1.FormBuilder }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.8", type: CheckboxInputComponent, selector: "lib-checkbox-input", inputs: { checkboxControl: "checkboxControl" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: CheckboxInputComponent,
                multi: true,
            },
        ], ngImport: i0, template: "<div [formGroup]=\"form\" style=\"margin-top: 10px\">\n  <div\n    *ngFor=\"let option of checkboxControl?.options; let idx = index\"\n    [formArrayName]=\"checkboxControl?.name\"\n  >\n    <mat-checkbox\n      [formControlName]=\"idx\"\n      [value]=\"option.value\"\n      (change)=\"onCheckboxChange($event, option, idx)\"\n    >\n      {{ option?.label }}\n    </mat-checkbox>\n  </div>\n</div>\n", styles: [""], dependencies: [{ kind: "directive", type: i2.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgControlStatusGroup, selector: "[formGroupName],[formArrayName],[ngModelGroup],[formGroup],form:not([ngNoForm]),[ngForm]" }, { kind: "directive", type: i1.FormGroupDirective, selector: "[formGroup]", inputs: ["formGroup"], outputs: ["ngSubmit"], exportAs: ["ngForm"] }, { kind: "directive", type: i1.FormControlName, selector: "[formControlName]", inputs: ["formControlName", "disabled", "ngModel"], outputs: ["ngModelChange"] }, { kind: "directive", type: i1.FormArrayName, selector: "[formArrayName]", inputs: ["formArrayName"] }, { kind: "component", type: i3.MatCheckbox, selector: "mat-checkbox", inputs: ["aria-label", "aria-labelledby", "aria-describedby", "id", "required", "labelPosition", "name", "value", "disableRipple", "tabIndex", "color", "checked", "disabled", "indeterminate"], outputs: ["change", "indeterminateChange"], exportAs: ["matCheckbox"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.8", ngImport: i0, type: CheckboxInputComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lib-checkbox-input', providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: CheckboxInputComponent,
                            multi: true,
                        },
                    ], template: "<div [formGroup]=\"form\" style=\"margin-top: 10px\">\n  <div\n    *ngFor=\"let option of checkboxControl?.options; let idx = index\"\n    [formArrayName]=\"checkboxControl?.name\"\n  >\n    <mat-checkbox\n      [formControlName]=\"idx\"\n      [value]=\"option.value\"\n      (change)=\"onCheckboxChange($event, option, idx)\"\n    >\n      {{ option?.label }}\n    </mat-checkbox>\n  </div>\n</div>\n" }]
        }], ctorParameters: () => [{ type: i1.FormBuilder }], propDecorators: { checkboxControl: [{
                type: Input
            }] } });
